\documentclass[12pt,letterpaper]{report}
%\documentclass[12pt,Bold,letterpaper]{mcgilletdclass}
\usepackage{amsfonts}
\usepackage{amsmath}
\usepackage{amssymb}
\usepackage{apacite}
\usepackage{caption}
\usepackage{color}
\usepackage{enumitem}
\usepackage{epsfig}
\usepackage{framed}
\usepackage[letterpaper, margin=1in]{geometry}		% Margins should be 1 inch according to McGill requirements (https://www.mcgill.ca/gps/thesis/guidelines/preparation)
\usepackage{graphicx}
\usepackage{listings}
\usepackage{placeins}
\usepackage{setspace}
\usepackage{subcaption}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}

\input{def_mcgill}

%-----------------------------------------------------------------------------
% Student info
%-----------------------------------------------------------------------------

\SetTitle{\huge{Penalized Regression Methods for Interaction and Mixed-Effects Models with Applications to Genomic and Brain Imaging Data}}
\SetAuthor{Sahir Rai Bhatnagar}
\SetDegreeType{Doctor of Philosophy}
\SetDepartment{Department of Epidemiology, Biostatistics and Occupational Health}
\SetUniversity{McGill University}
\SetUniversityAddr{Montréal, Québec, Canada}
\SetThesisDate{July 2018}
\SetRequirements{A thesis submitted to McGill University in partial fulfillment of the requirements of the degree of Doctor of Philosophy}
\SetCopyright{Sahir Rai Bhatnagar 2018}

%-----------------------------------------------------------------------------
% Document stats here
%-----------------------------------------------------------------------------

\begin{document}

% Create title page
\maketitle

%-----------------------------------------------------------------------------
% Input any special commands below
%-----------------------------------------------------------------------------

% Conditional expression for faster build during "development cycle"
\newif\ifthesis
\thesistrue       % Final thesis version: Uncomment this line and comment the next one
%\thesisfalse    % Development cycle: Uncomment this line and comment the previous one (faster LaTeX compile)

% Final version? 
% This is used to add items only for final submission
\newif\iffinal
%\finaltrue
\finalfalse

\ifthesis
	% Only show these sections when we are building the 'thesis' version

	\begin{romanPagenumber}{2}
	
		\SetDedicationName{\MakeUppercase{Dedication}}%
		\SetDedicationText{This document is dedicated to the graduate students of the McGill University.}%
		\Dedication%
	
	
	%-----------------------------------------------------------------------------
	% Acknowledgements:
	%   Among other acknowledgements, the student is required to declare the extent to which assistance (paid or unpaid) has 
	%   been given by members of staff, fellow students, research assistants, technicians, or others in the collection of materials 
	%   and data, the design and construction of apparatus, the performance of experiments, the analysis of data, and the 
	%   preparation of the thesis (including editorial help).
    %   In addition, it is appropriate to recognize the supervision and advice given by the thesis supervisor(s) and advisors.
	%-----------------------------------------------------------------------------		
	\SetAcknowledgeText{
I am most grateful to Mathieu Blanchette and Rob Sladek for the supervision of this thesis, their advice and guidance not only in professional issues, but also in all other fundamental aspects.
Many thanks to my PhD Committee: Jerome Waldispuhl, Doina Precup, Guillaume Bourque, and Derek Ruths for their helpful comments and suggestions. 
\\
I like to tank Douglas Ruden, Adrian Platts and Louis Letourneau for their insight and contributions to SnpEff and SnpSift projects.
\\
I am grateful to Mark McCarthy, John Blangero and Mike Boehnke, and David Altshuler for their leadership in the T2D consortia.
Special thanks to Pierre Fontanillas, Tanya Teslovich, Alisa Manning, Goo Jun, Anubha Mahajan, Jason Flannick, Andrew Morris, and Manuel Rivas for their helpful discussions that were instrumental in different aspects of this collaborative project.
\\
I thank Fiona Cunningham, Will McLaren, and Kai Wang for their contributions to the VCF variant annotation standard as well as Sarah Hunt for her efforts on the GA4GH annotations specification.
}
	\Acknowledge	

	%-----------------------------------------------------------------------------
	% Preface and contributions
	% In the case of collaborative work presented in either a standard format or manuscript-based thesis, there must be an 
	% explicit statement of the contributions of all parties, including the student, in the Preface of the thesis. 
   %  The Preface of a Doctoral thesis must also include a statement clearly indicating those elements of the thesis that 
   %  are considered original scholarship and distinct contributions to knowledge. 
	%-----------------------------------------------------------------------------		
	\SetPrefaceText{
		\noindent\textbf{Manuscript 1:} P. Cingolani, R. Sladek, and M. Blanchette. ``BigDataScript: a scripting language for data pipelines." Bioinformatics 31.1 (2015): 10-16.
		For this paper, PC conceptualized the idea and performed the language design and implementation. RS \& MB helped in designing robustness testing procedures. PC, RS \& MB wrote the manuscript.
		\\
		\\
		\textbf{Manuscript 2:} P. Cingolani, A. Platts, M. Coon, T. Nguyen, L. Wang, S.J. Land, X. Lu, D.M. Ruden, et al. ``A program for annotating and predicting the effects of single nucleotide polymorphisms, snpeff: Snps in the genome of drosophila melanogaster strain $w^{1118}; iso-2; iso-3$". Fly, 6(2), 2012.
		In this paper, PC conceptualized the idea, implemented the program and performed testing.
		AP contributed several feature ideas, software testing and suggested improvements.
		XL, DR, SL, LW, TN, MC, LW performed mutagenesis and sequencing experiments.
		XL and DR performed the biological interpretation of the data.
		All authors contributed to the manuscript.
		\\
		\\
		\textbf{Manuscript 3:} P. Cingolani, R. Sladek, and M. Blanchette. ``A co-evolutionary approach for detecting epistatic interactions in genome-wide association studies". Ready for submission (data embargo restrictions).
		For this paper, PC designed the methodology under the supervision of MB and RS. PC implemented the algorithms. PC, RS \& MB wrote the manuscript. 
	}
	\Preface	


	%-----------------------------------------------------------------------------
	%         English Abstract
	%-----------------------------------------------------------------------------
	
	\SetAbstractEnText{ 
In high-dimensional (HD) data, where the number of covariates ($p$) greatly exceeds the number of observations ($n$), estimation can benefit from the bet-on-sparsity principle, i.e., only a small number of predictors are relevant in the response. This assumption can lead to more interpretable models, improved predictive accuracy, and algorithms that are computationally efficient. In genomic and brain imaging studies, where the sample sizes are particularly small due to high data collection costs, we must often assume a sparse model because there isn't enough information to estimate $p$ parameters. For these reasons, penalized regression methods such as the lasso and group-lasso have generated substantial interest since they can set model coefficients exactly to zero. 
In the penalized regression framework, many approaches have been developed for main effects. However, there is a need for developing interaction and mixed-effects models. Indeed, accurate capture of interactions may hold the potential to better understand biological phenomena and improve prediction accuracy since they may reflect important modulation of a biological system by an external factor. Furthermore, penalized mixed-effects models that account for correlations due to groupings of observations can improve sensitivity and specificity. 
This thesis is composed primarily of three manuscripts. The first manuscript describes a novel strategy called \texttt{eclust} for dimension reduction that leverages the effects of an exposure variable with broad impact on HD measures. With \texttt{eclust}, we found improved prediction and variable selection performance compared to methods that do not consider the exposure in the clustering step, or to methods that use the original data as features. We further illustrate this modeling framework through the analysis of three data sets from very different fields, each with HD data, a binary exposure, and a phenotype of interest.
In the second manuscript, we propose a method called \texttt{sail} for detecting non-linear interactions that automatically enforces the strong heredity property using both the $\ell_1$ and $\ell_2$ penalty functions. We describe a blockwise coordinate descent procedure for solving the objective function and provide performance metrics on both simulated and real data. 
The third manuscript develops a general penalized mixed model framework to account for correlations in genetic data due to relatedness called \texttt{ggmix}. Our method can accommodate several sparsity-inducing penalties such as the lasso, elastic net and group lasso and also readily handles prior annotation information in the form of weights. Our algorithm has theoretical guarantees of convergence and we again assess its performance in both simulated and real data. We provide efficient implementations of all our algorithms in open source software.
	}
	\AbstractEn
	
	%-----------------------------------------------------------------------------
	%         French Abstract
	%-----------------------------------------------------------------------------
	
	\SetAbstractFrText{ 
Il est aujourd'hui possible d'obtenir la s\'equence du g\'enome de grandes cohortes d'individus, et cette information est permet de faciliter l'identification de variations g\'en\'etiques li\'ees \`a des maladies complexes. 
Dans ma th\`ese, j'\'etudie les d\'efis informatiques et statistiques li\'es \`a l'analyse de grands ensembles de donn\'ees g\'enomiques. 
J'aborde trois aspects de l'analyse. 
Premi\`erement, afin d'analyser de grandes quantit\'es de donn\'ees provenant d'\'etudes g\'enomiques nous concevons un langage de programmation, BigDataScript, qui simplifie la cr\'eation de pipelines d'analyse de donn\'ees robustes et \'evolutives. 
Deuxi\`emement, nous cr\'eons deux m\'ethodes d'annotation et de classification de variantes g\'enomiques (SnpEff et SnpSift) qui aident \`a pr\'edire leur l'effet possible.
Enfin, nous abordons le probl\`eme de l'identification de liens entre les maladies g\'en\'etiques et les variantes qui les causent en proposant une m\'ethodologie qui combine l'information g\'en\'etique au niveau d'une la population avec informations \'evolutive afin d'augmenter la puissance statistique des \'etudes d'association consid\'erant les interactions \'epistatiques.
	}
	\AbstractFr
	
	%-----------------------------------------------------------------------------
	% Tables
	%-----------------------------------------------------------------------------
	
	%\renewcommand{\contentsname}{Table of contents}
	\tableofcontents 

	\LOFHeading{List of Figures and Tables}
	%\listoftables 
	\listoffigures
	
	\end{romanPagenumber}

\else
	% Skip all the previous sections
\fi

%-----------------------------------------------------------------------------
% Chapters
%-----------------------------------------------------------------------------

% McGill requires double spacing
\doublespacing

\ifthesis
	% Show all chapters on 'thesis' mode
	\include{chapters/intro}
	\include{chapters/bds}
	\include{chapters/snpeff}
	\include{chapters/gwas}
	\include{chapters/conclusions}
\else
	% Only show the chapter we are currently working on (faster LaTex build)
	\include{chapters/gwas}
\fi

%-----------------------------------------------------------------------------
% Appendices
%-----------------------------------------------------------------------------

\ifthesis
	% Show all appendices (if any)
\else
	% Skip all appendices and bibliography
\fi
	
%-----------------------------------------------------------------------------
% Bibliography. You need to run the following 
% command: 
%       bibtex thesis_mcgill
%-----------------------------------------------------------------------------
\bibHeading{References}
%\bibliographystyle{plain}
\bibliographystyle{apacite}
\bibliography{mcgilletd}		% File: mcgilletd.bib

\end{document}
